File: Difficulty: Easy/Delete Head of Linked List/delete-head-of-linked-list.java
/*
class Node
{
    int data;
    Node next;

    Node(int data)
    {
        this.data = data;
        this.next = next;
    }
}
*/

class Solution {
    public Node deleteHead(Node head) {
        // code here
        if(head == null) return head;
        
        Node temp = head;
        head = head.next;
        
        return head;
    }
}

File: Difficulty: Easy/Delete Head of Linked List/delete-head-of-linked-list.cpp
/*
class Node {
public:
    int data;
    Node* next;

    Node(int x) {
        data = x;
        next = NULL;
    }
};

*/
class Solution {
  public:
    Node *deleteHead(Node *head) {
        // code here
        
        if(head == NULL) return head;
        
        Node *temp = head;
        head = head->next;
        
        free(temp);
        
        return head;
    }
};
